lib/matplotlib/projections/polar.py

```diff
@@ -1447,12 +1447,23 @@ def format_sig(value, delta, opt, fmt):
                     cbook._g_sig_digits(value, delta))
             return f"{value:-{opt}.{prec}{fmt}}"
 
-        return ('\N{GREEK SMALL LETTER THETA}={}\N{GREEK SMALL LETTER PI} '
-                '({}\N{DEGREE SIGN}), r={}').format(
+        # In case fmt_xdata was not specified, resort to default
+        if self.fmt_xdata is None:
+            return ('\N{GREEK SMALL LETTER THETA}={}\N{GREEK SMALL LETTER PI} '
+                    '({}\N{DEGREE SIGN}), r={}').format(
                     format_sig(theta_halfturns, delta_t_halfturns, "", "f"),
                     format_sig(theta_degrees, delta_t_degrees, "", "f"),
-                    format_sig(r, delta_r, "#", "g"),
+                    format_sig(r, delta_r, "#", "g")
+                    if self.fmt_ydata is None
```

Comment:
This is always true and can be removed, because you checked in l. 1451.

Reply:
Why does it have to be always true? In the if-statement we only check if `self.fmt_xdata` is not None. We can set `self.fmt_xdata = function` and `self.fmt_ydata` will remain None.

Reply:
🐑 sorry for the noise (mobile screens are definitely too small for a reasonable review).